src/CSnakes.Runtime/PyObjectTypeConverter.Generator.cs

```diff
@@ -0,0 +1,18 @@
+using CSnakes.Runtime.Python;
+using System.ComponentModel;
+using System.Globalization;
+using System.Reflection;
+
+namespace CSnakes.Runtime;
+internal partial class PyObjectTypeConverter
+{
+    private object? ConvertToGeneratorIterator(PyObject pyObject, Type destinationType, ITypeDescriptorContext? context, CultureInfo? culture)
+    {
+        Type item1Type = destinationType.GetGenericArguments()[0];
+        Type item2Type = destinationType.GetGenericArguments()[1];
+        Type item3Type = destinationType.GetGenericArguments()[2];
+        Type generatorType = typeof(GeneratorIterator<,,>).MakeGenericType(item1Type, item2Type, item3Type);
+        ConstructorInfo ctor = generatorType.GetConstructors().First();
+        return ctor.Invoke([pyObject.Clone()]);

```

Comment:
This is very slow in practice. Any chance you can make this strongly typed? I assume that you can get the actual types from the Python script?

Reply:
Will address in TypeConverter refactor :+1: